### ISSUE TITLE
Make B+ Tree Persistent

### ISSUE THREAD

**Issue by CinematicCow** [CinematicCow/Lumora#4]
Need to make the B+ tree persistent to save memory. Proposed solution :
- use [fs2](github.com/timtadh/fs2) package.

**Comment by CinematicCow** [comment]
Using the `fs2` package, we can change the init function as follows:

```go
func NewBPlusTree(filePath string) (*BPlusTree, error) {
	bf, err := fmap.CreateBlockFile(filePath)
	if err != nil {
		return nil, err
	}
	tree, err := bptree.New(bf, 8, -1)
	if err != nil {
		return nil, err
	}
	return &BPlusTree{tree: tree}, nil
}
```

Need to test with bench

**Comment by CinematicCow** [comment]
closed with 992bc83
